VLI-733: No dialog flow (#25)
vippsas/vipps-login-api: vipps-login-api.md

We recommend that apps initiate Vipps login in a webview. SafariViewController and Chrome Custom Tabs are preferred as these webviews are able to utilize cookies stored in the user's browser.

#### No dialog flow - check if the user is logged in

> This flow enables attempting a login with no interaction required by the user. This can be used to check if the user is already logged in. The user will always be returned to the `redirect_uri`.

Reviewer: ```suggestion
The no dialog flow attempts a login with no user interaction and is intended to be used to check if a user has already logged in. Completing this flow the user will always be returned to the `redirect_uri`.
```
Usikker på siste setningen, føles litt overflødig

> #### No dialog flow - check if the user is logged in
> This flow enables attempting a login with no interaction required by the user. This can be used to check if the user is already logged in. The user will always be returned to the `redirect_uri`.

Author: ```suggestion
#### No dialog flow - check if the user is logged in
This flow can be used as a method to check for existing authentication and/or consent. No interaction will be asked from the user in this flow.
```

Author: @IsakBjorn Nytt forslag på denne

Reviewer: Likte den!

> The user will be logged in with this flow if

Reviewer: ```suggestion
The user will be logged in with this flow if:
```

```
This operation does not require authentication

### Using the no dialog flow

> Simply add the query parameter `login_hint=unsolicited:nodialog` to the [Authorization request](#OAuth-2.0-Authorize) uri.

Reviewer: ```suggestion
To attempt a no dialog login add the query parameter `login_hint=unsolicited:nodialog` to the [Authorization request](#OAuth-2.0-Authorize) uri.
```

> The user will always be returned to the `redirect_uri`.

Reviewer: ```suggestion
```

> If the user logged in they will be returned to the `redirect_uri` with a code that can be used to complete the login. Just like in a regular [Authorization code](#authorization-code-grant) login.

Reviewer: ```suggestion
If the user completes the login they will be returned to the `redirect_uri` with a code that can be used to complete the login. Just like in a regular [Authorization code](#authorization-code-grant) login.
```

`https://client.example.com/callback?code=12adsa938721987321asd9873&state=123987218dh`

If the user is not logged in they will be returned with an error. Some possible errors are `interaction_required`, `login_required` or `server_error`. 

Not logged in return uri example: `https://client.example.com/callback?error=interaction_required&error_description=User+interaction+is+required&state=1312312321983212a3b`.

In all cases a new login can be started by removing the parameter `login_hint=unsolicited:nodialog` and initiating a new login for the user.

> ### Automatic Recognition

Reviewer: Slett dette punktet

Author: Slett hva?

Reviewer: Hele punktet om automatic recognition. Det handler om no dialog flowen og blir unødvendig av det du skriver nå
